File: TripKitAndroid/src/main/java/com/skedgo/tripkit/bookingproviders/BookingResolver.java
package com.skedgo.tripkit.bookingproviders;

import androidx.annotation.Nullable;

import com.skedgo.tripkit.BookingAction;
import com.skedgo.tripkit.ExternalActionParams;

import com.skedgo.tripkit.BookingAction;
import com.skedgo.tripkit.ExternalActionParams;
import io.reactivex.Observable;

public interface BookingResolver {
  int UBER = 0;
  int LYFT = UBER + 1;
  int FLITWAYS = LYFT + 1;
  int GOCATCH = FLITWAYS + 1;
  int INGOGO = GOCATCH + 1;
  int MTAXI = INGOGO + 1;
  int SMS = MTAXI + 1;
  int OTHERS = SMS + 1;

  Observable<BookingAction> performExternalActionAsync(ExternalActionParams params);
  @Nullable String getTitleForExternalAction(String externalAction);
}

File: TripKitAndroid/src/main/java/com/skedgo/tripkit/bookingproviders/BookingResolver.kt
package com.skedgo.tripkit.bookingproviders

import com.skedgo.tripkit.BookingAction
import com.skedgo.tripkit.ExternalActionParams
import io.reactivex.Observable

interface BookingResolver {
    fun performExternalActionAsync(params: ExternalActionParams): Observable<BookingAction>

    fun getTitleForExternalAction(externalAction: String): String?

    companion object {
        const val UBER: Int = 0
        const val LYFT: Int = UBER + 1
        const val FLITWAYS: Int = LYFT + 1
        const val GOCATCH: Int = FLITWAYS + 1
        const val INGOGO: Int = GOCATCH + 1
        const val MTAXI: Int = INGOGO + 1
        const val SMS: Int = MTAXI + 1
        const val OTHERS: Int = SMS + 1
    }
}